### ISSUE TITLE
IDBD meta-update NaN-poisons step-sizes past the clip guard (inf*0 through error*x*h)

### ISSUE THREAD

**Issue by ss251** [SlopDotCash/asi#42]
## Contribution provenance

- AI assistance: yes
- AI provider/model: anthropic / claude-fable-5 (verification, fix, report); xai / grok-4.6 (initial numerical-boundary sweep and reproductions, local lane)
- Client / agent tooling: claude-code; Grok Build CLI
- Attribution status: self-reported

**Describe the bug**

IDBD's meta-update NaN-poisons its state past the documented clip guard, permanently.

`log_alpha += beta * error * x * h` evaluates left-associative as `(error * x) * h`. With fresh traces (`h = 0`):

- a float32 overflow of a *finite* `|error * x|` (e.g. `error=1e20, x=1e20`) produces `inf`, then `inf * 0 = NaN`;
- an `inf` error (or observation) produces the same `0 * inf = NaN` directly.

The adjacent guard is commented "Clip log step-sizes to prevent numerical issues", but `jnp.clip(NaN, -10, 2)` is NaN — so `log_step_sizes`, `exp(log_alpha)`, traces, weight/bias deltas, and `bias_step_size` all go NaN and **stay NaN on every later finite update**. The gradient path (`update_from_gradient`, `z * traces`) has the identical hole.

**To Reproduce** (at `origin/main` `a0e3412`)

```
uv run python -c '
import jax.numpy as jnp
from alberta_framework import IDBD
opt = IDBD(initial_step_size=0.01, meta_step_size=0.01)
x = jnp.ones(2, dtype=jnp.float32); st = opt.init(2)
r = opt.update(st, jnp.array(jnp.inf, dtype=jnp.float32), x)
r2 = opt.update(r.new_state, jnp.array(1.0, dtype=jnp.float32), x)
rf = opt.update(st, jnp.array(1e20, dtype=jnp.float32), jnp.array([1e20, 1.0], dtype=jnp.float32))
print(r.new_state.log_step_sizes)   # [nan nan]
print(r2.new_state.log_step_sizes)  # [nan nan] — finite follow-up inherits it
print(rf.new_state.log_step_sizes)  # [nan -4.6051702] — only the overflowed channel
'
```

Controls: `error=0` and `error=-1` on the same init leave everything finite; the `x=1` channel of the overflow case stays at `log(0.01)`, so this is the overflowed product, not a global dtype issue.

**Expected behavior**

`h=0` means "no gradient correlation yet", so the meta-update must be 0 whenever `h` is 0 and the other factors are finite; the clip guard must leave `log_step_sizes` inside `[-10, 2]` (finite); and a later finite `(error, x)` must not inherit a permanently NaN step-size.

**Additional context**

Fix PR follows immediately: associate as `error * (x * h)` so a finite overflow dies at the zero trace, and complete the clip guard so a non-finite meta-delta skips adaptation for that weight (previous log kept) — same treatment on the bias twin and the `update_from_gradient` path, with failing-test-first regressions for all three plus an ordinary-adaptation-preserved pin.

Separate contract question, deliberately not bundled: `init(initial_step_size=1e-8)` stores `log ≈ -18.42`, and a zero-error first update still raises it to the clip floor `-10` (`alpha 1e-8 → 4.54e-5`) because the clip applies even when the meta-gradient is identically 0. If the clip floor is meant to bound *adaptation* rather than re-round legitimate initializations, that is a different, direction-needing change.
